frontend/src/App.tsx

```diff
@@ -252,13 +255,56 @@ function AppContainer() {
 
   React.useEffect(() => {
     if (desktopApi) {
-      desktopApi.receive(
-        'showReleaseNotes',
-        (data: { releaseNotes: string; appVersion: string }) => {
-          setReleaseNotes(data.releaseNotes);
-          setAppVersion(data.appVersion);
+      desktopApi.receive('appVersion', (currentBuildAppVersion: string) => {
+        setAppVersion(currentBuildAppVersion);
+        const octokit = new Octokit();
+
+        async function fetchRelease() {
+          const githubReleaseURL = `GET /repos/{owner}/{repo}/releases`;
+          // get me all the releases -> default decreasing order of releases
+          const response = await octokit.request(githubReleaseURL, {
+            owner: 'kinvolk',
+            repo: 'headlamp',
+          });
+          const latestRelease = response.data.find(
+            release => !release.name?.startsWith('headlamp-helm')
+          );
+          if (
+            latestRelease &&
+            semver.gt(latestRelease.name as string, appVersion) &&
+            !process.env.FLATPAK_ID
+          ) {
+            setReleaseDownloadURL(latestRelease.html_url);
+          }
+          /*
+    check if there is already a version in store if it exists don't store the current version
+    this check will help us later in determining whether we are on the latest release or not
+    */
+          const storedAppVersion = localStorage.getItem('app_version');
+          if (!storedAppVersion) {
+            localStorage.setItem('app_version', currentBuildAppVersion);
```

Comment:
Please abstract this in a function and replace this and the other call.

Reply:
done